src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -1,140 +1,112 @@
     const [recoveryCode, setRecoveryCode] = useState('');
-    const [needToClearError, setNeedToClearError] = useState(props.account.errors);
+    const [needToClearError, setNeedToClearError] = useState<Errors | boolean | null | undefined>(account?.errors);
 
-    const prevRequiresTwoFactorAuth = usePrevious(props.account.requiresTwoFactorAuth);
-    const prevValidateCode = usePrevious(props.credentials.validateCode);
+    const prevRequiresTwoFactorAuth = usePrevious(account?.requiresTwoFactorAuth);
+    const prevValidateCode = usePrevious(credentials?.validateCode);
 
-    const inputValidateCodeRef = useRef();
-    const input2FARef = useRef();
-    const timerRef = useRef();
+    const inputValidateCodeRef = useRef<MagicCodeInputHandle | null>();
+    const input2FARef = useRef<MagicCodeInputHandle | null>();
+    const timerRef = useRef<NodeJS.Timeout>();
 
-    const hasError = Boolean(props.account) && !_.isEmpty(props.account.errors) && !needToClearError;
-    const isLoadingResendValidationForm = props.account.loadingForm === CONST.FORMS.RESEND_VALIDATE_CODE_FORM;
-    const shouldDisableResendValidateCode = props.network.isOffline || props.account.isLoading;
+    const hasError = Boolean(account) && !isEmptyObject(account?.errors) && !needToClearError;
+    const isLoadingResendValidationForm = account?.loadingForm === CONST.FORMS.RESEND_VALIDATE_CODE_FORM;
+    const shouldDisableResendValidateCode = isOffline ?? account?.isLoading;
     const isValidateCodeFormSubmitting =
-        props.account.isLoading && props.account.loadingForm === (props.account.requiresTwoFactorAuth ? CONST.FORMS.VALIDATE_TFA_CODE_FORM : CONST.FORMS.VALIDATE_CODE_FORM);
+        account?.isLoading && account?.loadingForm === (account?.requiresTwoFactorAuth ? CONST.FORMS.VALIDATE_TFA_CODE_FORM : CONST.FORMS.VALIDATE_CODE_FORM);
 
     useEffect(() => {
-        if (!(inputValidateCodeRef.current && hasError && (props.session.autoAuthState === CONST.AUTO_AUTH_STATE.FAILED || props.account.isLoading))) {
+        if (!(inputValidateCodeRef.current && hasError && (session?.autoAuthState === CONST.AUTO_AUTH_STATE.FAILED || account?.isLoading))) {
             return;
         }
         inputValidateCodeRef.current.blur();
-    }, [props.account.isLoading, props.session.autoAuthState, hasError]);
+    }, [account?.isLoading, session?.autoAuthState, hasError]);
 
     useEffect(() => {
-        if (!inputValidateCodeRef.current || !canFocusInputOnScreenFocus() || !props.isVisible || !isFocused) {
+        if (!inputValidateCodeRef.current || !canFocusInputOnScreenFocus() || !isVisible || !isFocused) {
             return;
         }
         inputValidateCodeRef.current.focus();
-    }, [props.isVisible, isFocused]);
+    }, [isVisible, isFocused]);
 
     useEffect(() => {
-        if (prevValidateCode || !props.credentials.validateCode) {
+        if (!!prevValidateCode || !credentials?.validateCode) {
```

Comment:
If I remove it, eslint shows the following `Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator.`. I can obviously ignore it with a comment, the question is which do you think is better? Ignoring eslint or double negation?